### PR TITLE
Add modal window

### PR DESCRIPTION
**Pull request by AvinaashGhansam**

### CODE REVIEW

File: src/view/views/modal/SettingsWindow.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  };
+
+  const handleMusicButtonClick = () => {
+    setMusicEnabled((prev) => !prev);
+  };
+
+  const saveChangesAndClose = () => {
+    onClose();
+    setArButtonDisabled(true);
```
Comment (zlix12) [inline src/view/views/modal/SettingsWindow.tsx:55]: I'm not sure I understand why this is being done

File: src/view/views/modal/SettingsWindow.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  onClose,
+  muteARButton
+}: SettingsWindowProps) {
+  const { arEnabled, toggleAR } = useSetting();
+  const [isMusicEnabled, setMusicEnabled] = useState(false);
+  const [arButtonDisabled, setArButtonDisabled] = useState(false);
+
+  useEffect(() => {
+    setArButtonDisabled(!arEnabled);
+  }, [arEnabled]);
```
Comment (zlix12) [inline src/view/views/modal/SettingsWindow.tsx:41]: What's the necessity of this? Can you just base button states on `arEnabled`?

**Review by zlix12 — CHANGES_REQUESTED** [review]
Tests are failing because `SceneManager.test.tsx` renders using the three test renderer. You'll need to mock the components you added to `SceneControls`  because the three test renderer cannot render them.

File: package.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
     "@ar-js-org/ar.js": "^3.4.5",
     "@ar-js-org/artoolkit5-js": "^0.3.0",
     "@artcom/react-three-arjs": "^0.5.6",
+    "@chakra-ui/icons": "^2.1.1",
```
[the remark this answers is not in the document]
Comment (AvinaashGhansam) [reply package.json:24]: reverted to react-icons

[the remark this answers is not in the document]
Comment (AvinaashGhansam) [reply package.json:24]: resolved!

File: src/scene/SceneControls.tsx [hunk 4]
```diff
@@ -91,36 +92,48 @@ const SceneControls: React.FC<{
...
-      )}
-      <RenderIf
-        shouldRender={!!previousSceneTransition || !!nextSceneTransition}
+    <>
+      <Stack
+        padding={[2]}
+        direction="row"
+        position="absolute"
+        top={0}
+        right={2}
```
[the remark this answers is not in the document]
Comment (AvinaashGhansam) [reply src/scene/SceneControls.tsx:101]: resolved

File: src/scene/SceneControls.tsx [hunk 5]
```diff
@@ -91,36 +92,48 @@ const SceneControls: React.FC<{
...
       >
-        <Button
-          onClick={onRestart}
-          w="full"
-          colorScheme="gray"
-          isDisabled={isTransitioning}
+        {' '}
```
[the remark this answers is not in the document]
Comment (AvinaashGhansam) [reply src/scene/SceneControls.tsx:103]: removed!

File: src/view/views/modal/SettingsWindow.tsx (hunk 2 above)
Comment (AvinaashGhansam) [reply src/view/views/modal/SettingsWindow.tsx:41]: resolved

File: src/view/views/modal/SettingsWindow.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+              disabled={arButtonDisabled || muteARButton}
+            >
+              {arEnabled ? 'AR is ON' : 'AR is OFF'}
+            </Button>
+            <Button
+              colorScheme={isMusicEnabled ? 'green' : 'red'}
+              onClick={handleMusicButtonClick}
+              flex={1}
+            >
+              {isMusicEnabled ? 'Music is Enabled' : 'Music is Disabled'}
```
[the remark this answers is not in the document]
Comment (AvinaashGhansam) [reply src/view/views/modal/SettingsWindow.tsx:79]: resolved

File: src/view/views/modal/SettingsWindow.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        <ModalCloseButton />
+        <ModalBody>
+          <Flex justify="space-between">
+            <Button
+              colorScheme={arEnabled ? 'green' : 'red'}
+              onClick={handleARButtonClick}
+              flex={1}
+              disabled={arButtonDisabled || muteARButton}
+            >
+              {arEnabled ? 'AR is ON' : 'AR is OFF'}
```
[the remark this answers is not in the document]
Comment (AvinaashGhansam) [reply src/view/views/modal/SettingsWindow.tsx:72]: resolved

File: src/view/views/modal/SettingsWindow.tsx (hunk 1 above)
Comment (AvinaashGhansam) [reply src/view/views/modal/SettingsWindow.tsx:55]: refactored code to remove ambiguity

**Review by zlix12 — APPROVED** [review]

**Pull request merged**